### PR TITLE
kubeadm: switch UnversionedKubeletConfigMap to true

### PR DESCRIPTION
**Pull request by neolit123**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

- Graduate the feature gate to Beta and enable it by default.
- Pre-set the default value for UnversionedKubeletConfigMap
to "true" in test/e2e_kubeadm.
- Fix a couple of typos in "tolerate" introduced in the PR that
added the FG in 1.23.

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/kubeadm/issues/1582

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: graduate the UnversionedKubeletConfigMap feature gate to Beta and enable the feature by default. This implies that 1) for new clusters kubeadm will start using the "kube-system/kubelet-config" naming scheme for the kubelet ConfigMap and RBAC rules, instead of the legacy "kubelet-config-x.yy" naming. 2) during upgrade, kubeadm will only write the new scheme ConfigMap and RBAC objects. To disable the feature you can pass "UnversionedKubeletConfigMap: false" in the kubeadm config for new clusters. For upgrade on existing clusters you can also override the behavior by patching the ClusterConfiguration object in "kube-system/kubeadm-config". More details in the associated KEP.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://git.k8s.io/enhancements/keps/sig-cluster-lifecycle/kubeadm/2915-kubeadm-replace-kubelet-config-x.y
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/kubeconfig/kubeconfig_test.go [hunk 1]
```diff
@@ -484,7 +484,7 @@ func TestValidateKubeConfig(t *testing.T) {
 	configWithAnotherServerURL := setupdKubeConfigWithClientAuth(t, caCert, caKey, "https://4.3.2.1:4321", "test-cluster", "myOrg1")
 
 	// create a valid config but with whitespace around the CA PEM.
-	// validateKubeConfig() should tollerate that.
+	// validateKubeConfig() should tolerate that.
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/kubeconfig/kubeconfig_test.go:487]: left over typo fixes https://github.com/kubernetes/kubernetes/pull/105741/files#r740812762

**Comment by pacoxu** [comment]
> added [f6fbf6c](https://github.com/kubernetes/kubernetes/commit/f6fbf6c6e6dead3a9c8c834f24d1a993f77bf90f) to cleanup some mentions of the legacy naming.

/lgtm

**Comment by neolit123** [comment]
let's merge this to let it run in CI.
/hold cancel
/retest

**Pull request merged**